File: bin/alt-tab-send.lua
#!/usr/bin/lua5.3
local socket = require 'socket'
socket.unix = require 'socket.unix'

local s = assert(socket.unix())
local m = arg[1] or 'next\n'

assert(s:connect(os.getenv("SESSION_DIR").."/alt-tab-daemon"))
assert(s:send(m))

File: bin/alt-tab-send.py
#!/usr/bin/python3

from os import environ, path
from sys import argv
from socket import socket, AF_UNIX

SOCKET = path.join(environ['SESSION_DIR'], 'alt-tab-daemon')


def main(action: bytes):
    sock = socket(AF_UNIX)
    sock.connect(SOCKET)
    sock.send(action)


if __name__ == '__main__':
    if len(argv) > 1:
        action = argv[1]
    else:
        action = 'next'

    main(action.encode())
